### ISSUE TITLE
How I can contribute new rule 

### ISSUE THREAD

**Issue by deser** [benmosher/eslint-plugin-import#1279]
Hello, I've created https://github.com/deser/eslint-plugin-import-force-abbr pluging and want to suggest to include the rule inside into this ecosystem. Does it make sense to you, maintainers?

**Comment by ljharb** [comment]
Creating a separate plugin is putting it in the ecosystem :-)

Can you talk about the rationale for the rule, and it’s use cases?

**Comment by deser** [comment]
Yes, please look at https://github.com/deser/eslint-plugin-import-force-abbr/blob/master/docs/rules/abbr.md

**Comment by deser** [comment]
Yes, sure.
In our team we had modules (which are typically constants) which are used in a lot of places. As it is constant it is not always convenient to import utilizing destructuring (like `import { BLUE } from 'pencils'`) as these constant names could intersect with the same names but in different constants (`import {BLUE} from 'pens'`). Therefore we agreed that importing all constants to some namespace variable is convenient. But then we need to stick to the contract: a particular module should be always imported to the same variable name to exclude confusion.

**Comment by ljharb** [comment]
OK - why do you need that? `import { BLUE as BluePen } from 'pens'; import { BLUE as BluePencil } from 'pencils'` works just fine on a per-file basis.

**Comment by deser** [comment]
>>it seems like `moduleNameSubstr` is a very loose match; what if you have multiple modules with the same substring

This is covered, under the hood there is regex 
```
const getModuleSearchRegex = (str) => {
      if (!moduleRegexCache[str]) {
        moduleRegexCache[str] = new RegExp(`\\b${str}\\b(\\.[^.\\/:*?"<>|\r\n]+$|$)`, 'ig'); // the whole word at the end of the string
      }

      return moduleRegexCache[str];
    };
```

I agree that name is not correct and there are other issues to fix: https://github.com/deser/eslint-plugin-import-force-abbr/issues

**Comment by deser** [comment]
Also good use case is `import * as _ from 'lodash';`.

**Comment by deser** [comment]
@ljharb, what your final thoughts?
:)

**Comment by ljharb** [comment]
I think it might be worth it to have a rule that works for both import and require, and that's configured with a fully resolvable module path (not a substring or a regex) *as well* as with a way to specify the required binding name for a default import (or for the variable a require is stored in), as well as a required binding name for given named imports (by name).
